project1/src/main/java/uk/co/cpsd/javaproject1/Goat.java

```diff
@@ -47,7 +69,40 @@ public boolean decreaseEnergy(int currentTick){
             return energyLevel <= 0;
         }
 
-        System.out.println("Goat " + animalId + " energy decreased to " + energyLevel);
+        // System.out.println("Goat " + animalId + " energy decreased to " + energyLevel);
         return false;
     }
+
+    @Override
+    public Animal tryReproduceWith(Animal partner) {
+        if (!(partner instanceof Goat)) return null;
+        Goat other = (Goat) partner;
+
+        boolean samePosition = this.x == other.x && this.y == other.y;
+        boolean differentGender = this.gender != other.gender;
+        boolean bothEnergetic = this.energyLevel >= 20 && other.energyLevel >= 20;
```

Comment:
Probably worth another constant for energyToBreed.  Not least because if all of these (including things like tendency to move, and other bits) become attributes of each animal, you can regard them as their genetic makeup.  And when animals breed you can merge their genetic makeup with a random tweak.  And then you have evolution!